larky/src/main/java/com/verygood/security/larky/modules/vgs/vault/spi/LarkyVault.java

```diff
@@ -7,7 +7,7 @@
 
 public interface LarkyVault extends StarlarkValue {
 
-    Object redact(Object value, Object storage, Object format, List<Object> tags) throws EvalException;
+    Object redact(Object value, Object storage, Object format, List<Object> tags, Object decoratorConfig) throws EvalException;
```

Comment:
Should we move this to a builder pattern instead? Or do we just want to continue expanding the parameters? Does "decoratorConfig" really belong as a parameter ?

Reply:
I believe it makes API simpler to have a decorator as a parameter.
1. Builder adds some notion of state inside a script, complicating things.
2. You don't need to instantiate different objects if you have to tokenize different PII entities using different approaches:
``` python
load('@stdlib/json', 'json')
load('@stdlib//builtins', 'builtins')
load('@vgs//vault', 'vault')

def process(input, ctx):
    json_body = json.loads(str(input))
    
    json_body['card_number'] = vault.redact(json_body['card_number'], decoratorConfig=...)
    json_body['ssn'] = vault.redact(json_body['ssn'])
    return json.dumps(json_body)
```
3. Explicit is better than implicit.